### DISCUSSION TITLE
SymbolLoadingException

### DISCUSSION THREAD

**Discussion by courgeon** [dotnet/Silk.NET#465] [Q&A] [answered]
Hello again,

I'm running into a Runtime problem (Windows 10 - VS2019 pro )

Here is my code

            var glfw = Glfw.GetApi();
            if (!glfw.Init())
                throw new Exception("Unable to initialize GLFW");

            var gl = GL.GetApi(glfw.Context);
            string v = gl.GetStringS(StringName.Version);

However, on the last line, I get a Runtime exception :  

            Silk.NET.Core.Loader.SymbolLoadingException : 'Native symbol not found (Symbol: glGetString)'

I tried openning a window first with GLFW, and creating the context in various ways, but no luck.
Any suggestions ?

Thanks !

**Comment by courgeon** [comment]
Anyone with a working example of Silk.NET GLFW + OpenGL ?

**Comment by Perksey** [comment, marked as answer]
```cs
var gl = GL.GetApi(glfw.Context);
````

You're trying to use the context we use to load GLFW to load OpenGL functions, which won't work. Instead, you need to create a context which hooks into `glfwGetProcAddress`. Luckily, we've already done this for you. Try this line:

```cs
var gl = GL.GetApi(glfw, myCoolWindowPointer);
```

Where myCoolWindowPointer is obtained by `glfw.CreateWindow`. You can't create a `GlfwContext` until you have a window on-screen.

**Reply by courgeon** [reply]
I was looking for something like that, but using the Silk.NET.OpenGL on nuget (v2.3) there are no GL.GetApi that accepts two parameters. 
Your code thus can't compile. Is there another version ?

**Reply by Perksey** [reply]
Oh whoops, I missed out a crucial step. Here's the new code:
```cs
var gl = GL.GetApi(new GlfwContext(glfw, myCoolWindowPointer));
```

**Reply by courgeon** [reply]
Ok, here is my new code then 
`               var glfw = Glfw.GetApi();
                if (!glfw.Init())
                    throw new System.Exception("Unable to initialize GLFW");
                
                var glfwWindow = glfw.CreateWindow(800, 600, "Test", null, null);
                
                var gl = GL.GetApi(new GlfwContext(glfw, glfwWindow));
                
                string data = gl.GetStringS(StringName.Version);

                glfw.ShowWindow(glfwWindow);

                while (!glfw.WindowShouldClose(glfwWindow))
                {
                    glfw.PollEvents();
                    glfw.SwapBuffers(glfwWindow);
                }
`

But I still get the same error

      Silk.NET.Core.Loader.SymbolLoadingException : 'Native symbol not found (Symbol: glGetString)'

**Reply by Perksey** [reply]
Before the call use glfw.SetErrorCallback to find out what's happening

**Reply by courgeon** [reply]
Thank you for you help, it's really appreciated.

It does trigger the error callback

ErrorCode : NoContext
Description : Cannot query entry point without a current OpenGL or OpenGL ES context

**Reply by courgeon** [reply]
By the way, the window does popup. So GLFW seems to work fine. (if the GL calls are removed)

**Reply by courgeon** [reply]
Ok, figured it out, the context must be made current before creating the API

                var context = new GlfwContext(glfw, glfwWindow);
                context.MakeCurrent(); 
                var gl = GL.GetApi(context);

Thank you again !

**Reply by Perksey** [reply]
Ah yeah I forgot that Silk.NET.Windowing does that implicitly. Awesome!
